starlette/middleware/authentication.py

```diff
@@ -11,10 +12,20 @@
 from starlette.types import ASGIApp, ASGIInstance, Receive, Scope, Send
 
 
+def default_on_error(request: Request, exc: Exception):
+    return PlainTextResponse(str(exc), status_code=400)
+
+
 class AuthenticationMiddleware:
-    def __init__(self, app: ASGIApp, backend: AuthenticationBackend) -> None:
+    def __init__(
+        self,
+        app: ASGIApp,
+        backend: AuthenticationBackend,
+        on_error: typing.Callable = None,
+    ) -> None:
         self.app = app
         self.backend = backend
+        self.on_error = on_error if on_error is not None else default_on_error
```

Comment:
Could we have the default handler be a method on the class?
`self.on_error = on_error if on_error is not None else self.default_on_error`